lib/oeqa/selftest/cases/testutils.py

```diff
@@ -60,32 +61,38 @@ def qemu_bake_image(imagename):
     bitbake(imagename)
 
 
-def qemu_send_command(port, command, timeout=60):
+def qemu_send_command(port, command, timeout=120):
     command = ['ssh -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@localhost -p ' +
                str(port) + ' "' + command + '"']
     s2 = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     stdout, stderr = s2.communicate(timeout=timeout)
     return stdout, stderr, s2.returncode
 
 
+def metadir():
+    # Assume the directory layout for finding other layers. We could also
+    # make assumptions by using 'show-layers', but either way, if the
+    # layers we need aren't where we expect them, we are out of luck.
+    path = os.path.abspath(os.path.dirname(__file__))
+    metadir = path + "/../../../../../"
+
+    return metadir
+
+
 def akt_native_run(testInst, cmd, **kwargs):
     # run a command supplied by aktualizr-native and checks that:
     # - the executable exists
     # - the command runs without error
-    # NOTE: the base test class must have built aktualizr-native (in
-    # setUpClass, for example)
-    bb_vars = get_bb_vars(['SYSROOT_DESTDIR', 'base_prefix', 'libdir', 'bindir'],
-                          'aktualizr-native')
-    sysroot = bb_vars['SYSROOT_DESTDIR'] + bb_vars['base_prefix']
-    sysrootbin = bb_vars['SYSROOT_DESTDIR'] + bb_vars['bindir']
-    libdir = bb_vars['libdir']
-
-    program, *_ = cmd.split(' ')
-    p = '{}/{}'.format(sysrootbin, program)
-    testInst.assertTrue(os.path.isfile(p), msg="No {} found ({})".format(program, p))
-    env = dict(os.environ)
-    env['LD_LIBRARY_PATH'] = libdir
-    result = runCmd(cmd, env=env, native_sysroot=sysroot, ignore_status=True, **kwargs)
```

Comment:
Are you sure it's safe to remove the LD_LIBRARY_PATH bit? I seem to recall that was necessary for some obscure reason. If only I'd left a comment...

Reply:
I actually had problem with .so before this change. It looks like we had to do that because the library path was actually outside of the sysroot. With `run_cmd(..., native_sysroot=...)` and the bitbake trick from there https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/lib/oeqa/selftest/cases/package.py?id=2a6094ca8c4aa5c00636df77669e5bca6ad52bcf#n29, it seems that it works with less confusion.

Might need a double check on rocko and others though.

Reply:
Cool, no objection if it works!